File: src/lib/scala/Slate.Core/src/main/scala/slate/core/apis/ApiArg.scala
/**
<slate_header>
  author: Kishore Reddy
  url: https://github.com/kishorereddy/scala-slate
  copyright: 2015 Kishore Reddy
  license: https://github.com/kishorereddy/scala-slate/blob/master/LICENSE.md
  desc: a scala micro-framework
  usage: Please refer to license on github for more info.
</slate_header>
  */

package slate.core.apis


case class ApiArg (name: String = "", desc:String = "", required:Boolean = true,
                   defaultVal:String = "", eg:String = "" )
  extends scala.annotation.StaticAnnotation
{

}

File: src/lib/kotlin/slatekit-apis/src/main/kotlin/slatekit/apis/ApiArg.kt
/**
 * <slate_header>
 * url: www.slatekit.com
 * git: www.github.com/code-helix/slatekit
 * org: www.codehelix.co
 * author: Kishore Reddy
 * copyright: 2016 CodeHelix Solutions Inc.
 * license: refer to website and/or github
 * about: A tool-kit, utility library and server-backend
 * mantra: Simplicity above all else
 * </slate_header>
 */

package slatekit.apis

/**
 * Annotation to describe a parameter to an api action.
 * NOT CURRENTLY USED - Will be in upcoming versions.
 *
 * @param name : name of the argument
 * @param desc : description of argument
 * @param required : whether its required or not
 * @param defaultVal: 
 * @param eg
 */
@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.SOURCE)
@Repeatable
annotation class ApiArg(
    val name: String = "",
    val desc: String = "",
    val required: Boolean = true,
    val defaultVal: String = "",
    val eg: String = ""
)